libraries/chain/chain_controller.cpp

```diff
@@ -484,7 +489,7 @@ void chain_controller::_apply_block(const signed_block& next_block)
 
 } FC_CAPTURE_AND_RETHROW( (next_block.block_num()) )  }
 
-void chain_controller::check_transaction_authorization(const SignedTransaction& trx)const {
+void chain_controller::check_transaction_authorization(const SignedTransaction& trx, bool allow_unused_signatures)const {
    if ((_skip_flags & skip_transaction_signatures) && (_skip_flags & skip_authority_check)) {
```

Comment:
these skip flags need to be moved into named methods that return true or false. Too many errors can occur by doing bit and/or boolean math in if statements.